sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -264,4 +265,9 @@ object StructType {
       case _ =>
         throw new SparkException(s"Failed to merge incompatible data types $left and $right")
     }
+  
+  private[sql] def mapFields(fields: Array[StructField]): Map[String, StructField] = {
```

Comment:
The name is implying what the function does rather than what it returns. 
If you feel 'fieldsMap' is clearer or more in-line with spark coding convention, then I'm happy to change it.